instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/tracer_middleware.rb

```diff
@@ -121,18 +121,11 @@ def request_span_attributes(env:)
               'http.method' => env['REQUEST_METHOD'],
               'http.host' => env['HTTP_HOST'] || 'unknown',
               'http.scheme' => env['rack.url_scheme'],
-              'http.target' => fullpath(env)
+              'http.target' => env['QUERY_STRING'].empty? ? env['PATH_INFO'] : "#{env['PATH_INFO']}?#{env['QUERY_STRING']}"
             }
-            attributes['http.user_agent'] = env['HTTP_USER_AGENT'] if env['HTTP_USER_AGENT']
-            attributes.merge(allowed_request_headers(env))
-          end
 
-          # e.g., "/webshop/articles/4?s=1":
-          def fullpath(env)
-            query_string = env['QUERY_STRING']
-            path = env['SCRIPT_NAME'] + env['PATH_INFO']
-
-            query_string.empty? ? path : "#{path}?#{query_string}"
+            attributes['http.user_agent'] = env['HTTP_USER_AGENT'] if env['HTTP_USER_AGENT']
+            attributes.merge!(allowed_request_headers(env))
```

Comment:
Honestly I thought `merge!` returned `nil`, but that must be some other bang-type method I was thinking of.

Reply:
Yeah I usually double check when I use anything with a bang.  Probably thinking of `compact!`

```
compact!
Removes nil elements from the array.

Returns nil if no changes were made, otherwise returns the array.
```

https://ruby-doc.org/core-2.4.1/Array.html#method-i-compact-21